syntax = "proto2";

package rv.protos;

import "google/protobuf/struct.proto";
import "rastervision/protos/raster_transformer2.proto";

message RasterSourceConfig {
    message GeoTiffFiles {
        repeated string uris = 1;
    }

    message ImageFile {
        required string uri = 1;
    }

    required string source_type = 1;
    repeated RasterTransformerConfig transformers = 2;
    // The channel indices and order to use when extracting chips from a
    // RasterSource. For instance, this should be an array of length 3 to
    // generate RGB chips.
    repeated int32 channel_order = 3;

    oneof raster_source_config {
        GeoTiffFiles geotiff_files = 4;
        ImageFile image_file = 5;
        google.protobuf.Struct custom_config = 6;
    }
}
